staging/src/k8s.io/apiextensions-apiserver/test/integration/testserver/resources.go

```diff
@@ -37,6 +37,26 @@ const (
 	noxuInstanceNum int64 = 9223372036854775807
 )
 
+// New generates a CRD with random name to avoid name conflict in e2e tests
+func NewNoxuCustomResourceDefinitionForAdmissionWebhookTest(scope apiextensionsv1beta1.ResourceScope) *apiextensionsv1beta1.CustomResourceDefinition {
+	// ensure the singular doesn't end in an s for now
+	gName := names.SimpleNameGenerator.GenerateName("foo") + "a"
```

Comment:
this looks like a flake waiting to happen.

Reply:
I thought we were going to do CRD in a second PR. I guess if this works I won't block it.